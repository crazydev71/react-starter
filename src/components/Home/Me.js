import React, { useEffect, useState } from 'react';

/**
 * Example component of using `useEffect` hook without lifecycle method
 *
 * IMPORTANT:
 * Make sure to pass empty array as param in use Effect.
 * This will make sure effet hook is called one time.
 * Otherwise it will be called every time component updated.
 */
const Me = () => {
  const [user, setUser] = useState(null);

  useEffect(() => {
    fetch('https://randomuser.me/api/')
      .then((results) => results.json())
      .then((data) => {
        setUser(data.results[0]);
      });
  }, []);

  return (
    <div className="me">
      <h2>*** Example for using useEffect! ***</h2>
      <h3>My random name: {user ? `${user.name.first} ${user.name.last}` : ''}</h3>
    </div>
  );
};

export default Me;
