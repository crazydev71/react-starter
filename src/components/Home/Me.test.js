import React from 'react';
import { shallow, mount } from 'enzyme';
import Me from './Me';

describe('Me component', () => {
  it('Should render Me component.', () => {
    const wrapper = shallow(<Me />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Mock api calling', (done) => {
    const mockUser = {
      name: { first: 'Some', last: 'Name' },
    };
    const mockJsonPromise = Promise.resolve({ results: [mockUser] });
    const mockFetchPromise = Promise.resolve({
      json: () => mockJsonPromise,
    });

    jest.spyOn(global, 'fetch').mockImplementation(() => mockFetchPromise);

    const wrapper = mount(<Me />);

    expect(global.fetch).toHaveBeenCalledTimes(1);
    expect(global.fetch).toHaveBeenCalledWith('https://randomuser.me/api/');

    process.nextTick(() => {
      expect(wrapper.text()).toContain(mockUser.name.first);
      expect(wrapper.text()).toContain(mockUser.name.last);

      global.fetch.mockClear();
      done();
    });
  });
});

jest.setTimeout(20 * 1000);
