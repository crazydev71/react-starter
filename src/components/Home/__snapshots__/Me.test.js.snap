// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Me component Should render Me component. 1`] = `
<div
  className="me"
>
  <h2>
    *** Example for using useEffect ***
  </h2>
  <h3>
    My random name: 
  </h3>
</div>
`;
